### PR TITLE
Prompush Reporter: Adding Option to pass the apikeyfile

### PR DESCRIPTION
**Pull request by sahankj2000**
## Description
- Making the changes in the endpoint format as suggested in #1983 
- Adding a new NBCLI option to be to pass the prompush_apikeyfile location via CLI rather than just having it in a predefined location (`~/.nosqlbench/prompush/prompush_apikey`)

### CODE REVIEW

File: nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/engine/metrics/reporters/PromPushReporterComponent.java [hunk 1]
```diff
@@ -50,37 +50,54 @@ public class PromPushReporterComponent extends PeriodicTaskComponent {
...
 
-        if (endpoint.matches("victoria:[a-zA-Z0-9._-]+:[0-9]+")) {
+        if (endpoint.matches("victoria:plain:[a-zA-Z0-9._-]+(:[0-9]+)?")) {
+            String[] parts = endpoint.split(":", 3);
+            endpoint = "http://" + parts[2] + "/api/v1/import/prometheus/metrics/job/JOBNAME/instance/INSTANCE";
+        } else if (endpoint.matches("victoria:tls:[a-zA-Z0-9._-]+(:[0-9]+)?")) {
+            String[] parts = endpoint.split(":", 3);
+            endpoint = "https://" + parts[2] + "/api/v1/import/prometheus/metrics/job/JOBNAME/instance/INSTANCE";
+        } else if (endpoint.matches("victoria:[a-zA-Z0-9._-]+(:[0-9]+)?")) {
+            System.out.println("match3");
```
Comment (dave2wave) [inline nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/engine/metrics/reporters/PromPushReporterComponent.java:68]: Either remove this debugging code or use a logger.

**Review by dave2wave — CHANGES_REQUESTED** [review]
A small change - otherwise LGTM

File: nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/engine/metrics/reporters/PromPushReporterComponent.java (hunk 1 above)
Comment (sahankj2000) [reply nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/engine/metrics/reporters/PromPushReporterComponent.java]: Sorry, missed that. Removed!!

**Review by dave2wave — APPROVED** [review]
LGTM

**Pull request merged**